Engines/Wine/Engine/Object/script.js

```diff
@@ -5,333 +5,362 @@ include("utils.functions.net.download");
-    return this;
-};
+    /**
+     * @param {String} [path]
+     * @returns {String}
+     */
+    winepath(path) {
+        return this.run("winepath", ["-w", path], this.prefixDirectory(), true, true);
+    }
 
-/**
- * @param {String} [path]
- * @returns {String}
- */
-Wine.prototype.winepath = function (path) {
-    return this.run("winepath", ["-w", path], this.prefixDirectory(), true, true);
-}
+    /**
+     *
+     * @param {string} [prefix]
+     * @param {string} [distribution]
+     * @param {string} [architecture]
+     * @param {string} [version]
+     * @returns {string|Wine}
+     */
+    prefix(prefix, distribution, architecture, version) {
+        // get
+        if (arguments.length == 0) {
+            return this._implementation.getWorkingContainer();
+        }
+
+        // set
+        else if (arguments.length == 1) {
+            this._implementation.setWorkingContainer(prefix);
+        } else {
+            const operatingSystem = this.operatingSystemFetcher.fetchCurrentOperationSystem().getWinePackage();
+            const subCategory = distribution + "-" + operatingSystem + "-" + architecture;
+
+            this._implementation.createContainer(subCategory, version, prefix);
+            this._implementation.setWorkingContainer(prefix);
+        }
 
-/**
-*
-* @param {string} [prefix]
-* @param {string} [distribution]
-* @param {string} [architecture]
-* @param {string} [version]
-* @returns {string|Wine}
-*/
-Wine.prototype.prefix = function (prefix, distribution, architecture, version) {
-    // get
-    if (arguments.length == 0) {
-        return this._implementation.getWorkingContainer();
-    }
-    // set
-    else if (arguments.length == 1) {
-        this._implementation.setWorkingContainer(prefix);
         return this;
     }
-    else {
-        var operatingSystem = this._OperatingSystemFetcher.fetchCurrentOperationSystem().getWinePackage();
-        var subCategory = distribution + "-" + operatingSystem + "-" + architecture;
-        this._implementation.createContainer(subCategory, version, prefix);
-        this._implementation.setWorkingContainer(prefix);
-        return this;
+
+    /**
+     * returns prefix directory
+     * @returns {string}
+     */
+    prefixDirectory() {
+        return this._implementation.getContainerDirectory(this._implementation.getWorkingContainer());
     }
-};
 
-/**
-* returns prefix directory
-* @returns {string}
-*/
-Wine.prototype.prefixDirectory = function () {
-    return this._implementation.getContainerDirectory(this._implementation.getWorkingContainer());
-};
+    /**
+     * returns the path to the engine binary directory
+     * if no parameters are given, the Wine version of the current prefix is used
+     * @param {string} [subCategory] Wine sub-category
+     * @param {string} [version] Wine version
+     * @returns {string} path to "wine" binary
+     */
+    binPath(subCategory, version) {
+        if (0 == arguments.length) {
```

Comment:
I don't like the `arguments.length == x` checks. Maybe we can find a better solution to do this?

Reply:
You can check if the parameter is defined if you prefer that.

Reply:
Let us retain it as it is for now